### ISSUE TITLE
Allow '.' in Anchors and Aliases

### ISSUE THREAD

**Issue by RavindraBarapatre** [aaubry/YamlDotNet#382]
Hello,

can we add support for '.' in Anchors and Aliases?
'.' is valid in YAML anchors and aliases. but while converting YAML to object. It is not allowing the '.' character.

Currently in YAMLDotNet, 0-9a-zA-Z_\- regex used to validate anchors and aliases.

**Comment by br3nt** [comment]
For completeness, the anchor property, `c-ns-anchor-property` is defined by the following:

```
[1]  c-printable ::=
    #x9 | #xA | #xD | [#x20-#x7E]            (8-bit)
    | #x85 | [#xA0-#xD7FF] | [#xE000-#xFFFD] (16-bit)
    | [#x10000-#x10FFFF]                     (32-bit)
[3]  c-byte-order-mark ::= #xFEFF            (see: http://www.unicode.org/unicode/faq/utf_bom.html)
[23] c-flow-indicator | ::= "," | "[" | "]" | "{" | "}"
[24] b-line-feed ::= #xA                     (LF)
[25] b-carriage-return ::= #xD               (CR)
[26] b-char ::= b-line-feed | b-carriage-return
[27] nb-char ::= c-printable - b-char - c-byte-order-mark
[31] s-space ::= #x20                        (SP)
[32] s-tab ::= #x9                           (TAB)
[33] s-white ::= s-space | s-tab
[34] ns-char ::= nb-char - s-white
[102] ns-anchor-char ::= ns-char - c-flow-indicator
[103] ns-anchor-name ::= ns-anchor-char+
[101] c-ns-anchor-property ::= "&" ns-anchor-name
```

**Comment by aaubry** [comment]
This has been fixed by #403, and is included in release 6.1.1 - https://dotnetfiddle.net/Q2HCSY
